mypy/test/helpers.py

```diff
@@ -308,3 +249,52 @@ def retry_on_error(func: Callable[[], Any], max_wait: float = 1.0) -> None:
                 # Done enough waiting, the error seems persistent.
                 raise
             time.sleep(wait_time)
+
+
+class AssertionFailure(Exception):
```

Comment:
This Exception and the `assert*`s below are the targets I mentioned replacing. They entirely duplicate pytest functionality.

Reply:
Do you want me to do anything about this in current PR? I wanted to leave these changes to the following PRs.

Reply:
No, I don't think that is needed.